filebeat/harvester/registry.go

```diff
@@ -68,13 +72,12 @@ func (r *Registry) Start(h Harvester) {
 	}
 
 	r.wg.Add(1)
-	r.harvesters[h.ID()] = h
-
 	go func() {
 		defer func() {
 			r.remove(h)
 			r.wg.Done()
 		}()
+		r.add(h)
```

Comment:
this has a very high chance to insert the harvester into the registry, after Start did return. Can we run into a race with the prospector requiring an up-to-date registry?